kubectl-cloudflow/docker/inspect.go

```diff
@@ -0,0 +1,196 @@
+// This code part is copied and modified from the skopeo project,
+// https://github.com/containers/skopeo.
+package docker
+
+import (
+	"context"
+	"fmt"
+	"github.com/containers/image/v5/image"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+	"strings"
+	"time"
+)
+
+// This code part is copied and modified from the skopeo project,
+// https://github.com/containers/skopeo.
+type InspectOutput struct {
+	Name          string `json:",omitempty"`
+	Tag           string `json:",omitempty"`
+	Digest        digest.Digest
+	Created       *time.Time
+	DockerVersion string
+	Labels        map[string]string
+	Architecture  string
+	Os            string
+	Layers        []string
+	Env           []string
+}
+
+// Inspect execution specific options
+type InspectOptions struct {
+	commandTimeout time.Duration
+	Image  *ImageOptions
+}
+
+// ImageOptions collects info which may be different for each image
+type ImageOptions struct {
+	authFilePath string // Path to a */containers/auth.json
+	credsOption    optionalString      // username[:password] for accessing a registry
+	dockerCertPath string              // A directory using Docker-like *.{crt,cert,key} files for connecting to a registry or a daemon
+	tlsVerify      bool        // Require HTTPS and verify certificates (for docker: and docker-daemon:)
+	noCreds        bool                // Access the registry anonymously
+	sharedBlobDir    string // A directory to use for OCI blobs, shared across repositories
+	dockerDaemonHost string // docker-daemon: host to connect to
+}
+
+// optionalString is a string with a separate presence flag.
+type optionalString struct {
+	present bool
+	value   string
+}
+
+func (opts *InspectOptions) InspectLocalDockerImage(imageName string)(imageData *InspectOutput, retErr error) {
+	return opts.inspectImage("docker-daemon:" + imageName)
```

Comment:
maybe we can link this to a flag in the command.. (just an idea, 'verify local')

Reply:
Let's address this when we decide upon the UX. Not fix it here.

Reply:
See above. I would like to remove the local inspect unless we have a very good reason for it.